solr/core/src/java/org/apache/solr/search/facet/FacetProcessor.java

```diff
@@ -419,7 +419,7 @@ void fillBucket(SimpleOrderedMap<Object> bucket, Query q, DocSet result, boolean
       }
       count = result.size();  // don't really need this if we are skipping, but it's free.
     } else {
-      if (q == null) {
+      if (q == null || fcontext.base.size() == 0) {
```

Comment:
The query is already built at this point, so I don't think this particular change actually helps wrt SOLR-10732? (and the `base.size()==0` case is already trivially optimized in `SolrIndexSearcher.numDocs(Query, DocSet)`)

Reply:
Will remove this before committing

Reply:
This is done https://github.com/apache/lucene-solr/pull/2123/commits/c194e09ca0d2df32acf21875c7625f9e862fdc09